pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -136,6 +143,11 @@ public double toDouble(Object value) {
       return ((Number) value).doubleValue();
     }
 
+    @Override
+    public BigDecimal toBigDecimal(Object value) {
+      return new BigDecimal(toInt(value));
```

Comment:
Should we consider using `BigDecimal.valueOf()`? Same for other places

Reply:
Using `BigDecimal.valueOf()` requires extra cast from `int` to `long` and hence I use the constructor that takes int.

Reply:
I think it should be the same because internally the value is casted to `long`. For small int, we can reuse the object